test/e2e/pull_subscription.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"cloud.google.com/go/pubsub"
+	"context"
+	"fmt"
+	"knative.dev/pkg/test/helpers"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	ProwProjectKey = "E2E_PROJECT_ID" //"GCP_PROJECT"
+)
+
+func makeTopicOrDie(t *testing.T) string {
```

Comment:
An interesting pattern I saw in one of the hashicorp testing presentations is to have test helpers return their own 'undo-ers'. So this would become:

```golang
func makeTopicOrDie(t *testing.T) (string, func()) {
  ...
  return topicName, deleteTopicOrDieFunc(t, topicName)
}
```

Then the calling code:

```golang
  t, d := makeTopicOrDie(t)
  defer d
  ...
```
That way we make sure the defer is setup as soon as we can. Not applicable here, but it also can hide what is being created by never exposing its name.

Reply:
GREAT IDEA.